File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task16_GEMINI_gemini-1.5-pro-001.java
public class Solution {
  public static boolean solution(String str, String ending) {
    if (str.length() < ending.length()) {
      return false;
    }
    return str.substring(str.length() - ending.length()).equals(ending);
  }
  
  public static void main(String[] args) {
    System.out.println(solution("abc", "bc")); // true
    System.out.println(solution("abc", "d")); // false
  }
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task16_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <string>

bool solution(const std::string& str, const std::string& ending) {
  if (str.length() < ending.length()) {
    return false;
  }
  return str.find(ending, str.length() - ending.length()) != std::string::npos;
}

int main() {
  std::cout << solution("abc", "bc") << std::endl; // 1 (true)
  std::cout << solution("abc", "d") << std::endl; // 0 (false)
  return 0;
}